scrapy/resolver.py

```diff
@@ -1,14 +1,85 @@
-from twisted.internet import defer
+import socket
+from twisted.python import failure
+from twisted.internet import defer, threads
+from twisted.internet.error import DNSLookupError
 from twisted.internet.base import ThreadedResolver
 
 from scrapy.utils.datatypes import LocalCache
 
+
+IPV4_ONLY = 4
+IPV6_ONLY = 6
+IPV4_PREFERRED = 46
+IPV6_PREFERRED = 64
+
+
+class IPv6ThreadedResolver(ThreadedResolver):
+    """ThreadedResolver calling socket.getaddrinfo for IPv6 capability.
+
+    Uses a threadpool to perform name lookups without blocking the
+    reactor thread. The underlying socket.getaddrinfo is a blocking call.
+    """
+
+    def __init__(self, reactor, protocol_preference=46):
+        super(IPv6ThreadedResolver, self).__init__(reactor)
+        if protocol_preference == 0:
```

Comment:
The default for `settings.getint()` when the setting is unset, is `0`. In this case, the `socket.getaddrinfo` call seems to prefer IPv6. I thought best to be safe here.
And I added the setting to `default_settings.py` only afterwards.